pycbc/strain/recalibrate.py

```diff
@@ -512,3 +515,109 @@ def map_to_adjust(self, strain, **params):
                               kappa_pu_re=calib_args[6],
                               kappa_pu_im=calib_args[7])
         return strain_adjusted
+
+
+def read_calibration_envelop_file(calibration_file, correction_type,
+                                  minimum_frequency, maximum_frequency, n_nodes):
+    """
+    This function reads the calibration envelop file and provide arrays
+    needed to construct cubic splines
+
+    Input:
+      calibration_file: Calibration envelop file
+      correction_type: Specify the correction type between 'data' and 'template'
+      minimum_frequency: Lower cut-off frequency to apply calibration
+      maximum_frequency: Upper cut-off frequency to apply calibration
+      n_nodes: Nodal points for cubic-splines
+
+    Output:
+      log_nodes: Values of log-frequency for nodal points
+      amplitude_median_nodes: Median values of Amplitude correction at frequency nodal points
+      amplitude_sigma_nodes: Standard deviation values of Amplitude correction at frequency nodal points
+      phase_median_nodes: Median values of Phase correction at frequency nodal points
+      phase_sigma_nodes: Standard deviation values of Phase correction at frequency nodal points
+    """
+    calibration_data = np.loadtxt(calibration_file).T
+    log_frequency_array = np.log(calibration_data[0])
+
+    log_nodes = np.linspace(np.log(minimum_frequency),
+                            np.log(maximum_frequency), n_nodes)
+
+    if correction_type.lower()=='template':
+        amplitude_median = calibration_data[1] - 1
+        phase_median = calibration_data[2]
+        amplitude_sigma = abs(calibration_data[5] - calibration_data[3]) / 2
+        phase_sigma = abs(calibration_data[6] - calibration_data[4]) / 2
+
+    elif correction_type.lower()=='data':
+        amplitude_median = 1/calibration_data[1] -1
+        phase_median = -calibration_data[2]
+        amplitude_sigma = abs(1/calibration_data[3] - 1/calibration_data[5]) / 2
+        phase_sigma = abs(calibration_data[6] - calibration_data[4]) / 2
+
+    # Some sanity checks
+    if calibration_data[0][-1] < maximum_frequency:
+        if maximum_frequency - calibration_data[0][-1] < 0.5:
+            maximum_frequency = calibration_data[0][-1]
+        else:
+            raise ValueError("Maximum frequency (=%d) for tc=%s is more than"
+                         "maximum frequency in calibration envelop (=%.3g)"%(maximum_frequency,gps_time,calibration_data[0][-1]))
+    else:
+        pass
```

Comment:
I don't think this is necessary.

Reply:
In general, it might not be needed, but I still want to keep it in case we go beyond the frequency range of the calibration provided by the calibration envelopes. For example, if someone wants to do PE with a lower cutoff frequency than the standard, they might be tempted to give the same f_lower for calibration config file generation. This option should safeguard against that.

Reply:
Ah, just to clarify, I mean then `else` part of the `if/else` statement. I think it's a good idea to keep the checks you implemented.

Reply:
Got it. Thanks for the clarification. I agree with that. Will do in the following commit.